### PR TITLE
test: bsim: Notification EATT test

### PR DESCRIPTION
**Pull request by CarbuneanuAlexandru**
Increasing the test coverage of notification on EATT bearers.

Signed-off-by: Alexandru Carbuneanu <alexandru.carbuneanu@nordicsemi.no>

### CODE REVIEW

**Review by jhedberg — CHANGES_REQUESTED** [review]
This PR needs to be split up into multiple commits. At the very least the stack changes (including public APIs) should be in a separate commit.

File: include/bluetooth/att.h [hunk 1]
```diff
@@ -65,6 +65,34 @@ size_t bt_eatt_count(struct bt_conn *conn);
 #endif /* CONFIG_BT_TESTING */
 #endif /* CONFIG_BT_EATT */
 
+<<<<<<< HEAD
```
Comment (alwa-nordic) [inline include/bluetooth/att.h]: Bad merge.

**Review by alwa-nordic — CHANGES_REQUESTED** [review]

**Comment by Thalley** [comment]
Looks like this builds on top of https://github.com/zephyrproject-rtos/zephyr/pull/43112 - In the future please do add reference to other PRs that it may build on, so it is clear for the reviewers which commits to review. 

Alternatively, only upload the changes you want reviewed, and have the PR fail on CI. 

You can also create PRs as `draft` so that you can run your CI etc., but without triggering any review requests :)

I'll review this PR once https://github.com/zephyrproject-rtos/zephyr/pull/43112 is merged.

**Comment by CarbuneanuAlexandru** [comment]
> Looks like this builds on top of #43112 - In the future please do add reference to other PRs that it may build on, so it is clear for the reviewers which commits to review.
> 
> Alternatively, only upload the changes you want reviewed, and have the PR fail on CI.
> 
> You can also create PRs as `draft` so that you can run your CI etc., but without triggering any review requests :)
> 
> I'll review this PR once #43112 is merged.

yes, I was not sure how to handle it, it needs changes from an other PR to work, so the commits to be reviewed here are only the last two once:
d1082a2160e2d3637348d3ba71163ab01b42121d
a9b46352389e2643a2dfb42eb1b6a2e31355d5b8

**Pull request closed without merging**